pkg/provider/aws/action/eks/eks.go

```diff
@@ -0,0 +1,268 @@
+// Just copied from AKS
+
+package eks
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi-aws/sdk/v6/go/aws/ec2"
+	"github.com/pulumi/pulumi-aws/sdk/v6/go/aws/eks"
+	"github.com/pulumi/pulumi-aws/sdk/v6/go/aws/iam"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/redhat-developer/mapt/pkg/manager"
+	maptContext "github.com/redhat-developer/mapt/pkg/manager/context"
+	"github.com/redhat-developer/mapt/pkg/provider/aws"
+	"github.com/redhat-developer/mapt/pkg/provider/util/output"
+	"github.com/redhat-developer/mapt/pkg/util/logging"
+)
+
+type EKSRequest struct {
+	Prefix   string
+	Location string
+	VMSize   string
+	KubernetesVersion string
+	ScalingDesiredSize int
+	ScalingMaxSize     int
+	ScalingMinSize     int
+	Addons 	[]string
+	// OnlySystemPool    bool
+	// Spot              bool
+	// SpotTolerance     spotAzure.EvictionRate
+}
+
+func Create(ctx *maptContext.ContextArgs, r *EKSRequest) (err error) {
+	logging.Debug("Creating EKS")
+	if err := maptContext.Init(ctx, aws.Provider()); err != nil {
+		return err
+	}
+	cs := manager.Stack{
+		StackName:           maptContext.StackNameByProject(stackName),
+		ProjectName:         maptContext.ProjectName(),
+		BackedURL:           maptContext.BackedURL(),
+		ProviderCredentials: aws.DefaultCredentials,
+		DeployFunc:          r.deployer,
+	}
+	sr, _ := manager.UpStack(cs)
+	return r.manageResults(sr)
+}
+
+func Destroy(ctx *maptContext.ContextArgs) error {
+	// Create mapt Context
+	logging.Debug("Destroy EKS")
+	if err := maptContext.Init(ctx, aws.Provider()); err != nil {
+		return err
+	}
+	return aws.DestroyStack(
+		aws.DestroyStackRequest{
+			BackedURL: maptContext.BackedURL(),
+			Stackname: stackName,
+	})
+}
+
+// Main function to deploy all requried resources to AWS
+func (r *EKSRequest) deployer(ctx *pulumi.Context) error {
+	// Read back the default VPC and public subnets, which we will use.
+	t := true
+	vpc, err := ec2.LookupVpc(ctx, &ec2.LookupVpcArgs{Default: &t})
```

Comment:
You can use https://github.com/redhat-developer/mapt/blob/main/pkg/provider/aws/action/fedora/fedora.go#L192 to create the networking, typically each action create its own stack of networking as that will not increase cost and it will allow to isolate the service

Reply:
I tried to use it now, but I encountered an issue where the EKS cluster requires at least two subnets in two availability zones, but the current implementation of networking uses only one. I tried to extend it, but I got stuck on this as it became increasingly complex. I created the `network_extended/network_extended.go` based on `network.go` for trying to implement it (because editing `network.go` kept breaking other things that use it).

Reply:
From this: https://github.com/redhat-developer/mapt/pull/472/files#diff-4593130eb8793764853fb6763e38a250579d0ca6a8a3f3b7c6681782cdd5b842R85-R88 I always get only one subnet 🤔

Reply:
With the networking, I figured out that I need to define multiple CIDRs, but now I'm not sure how to attach the subnets to the VPC (error: * updating EKS Cluster (eks-cluster-5370d02) VPC configuration: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: 591ad7dc-961f-46b9-804b-21498734fb05, InvalidRequestException: Subnets specified must belong to the VPC: vpc-03ae06bf486414612 ), pushed a [commit](https://github.com/zdrapela/mapt/commit/c37f6bfa0a70409f17aae56247ec670dd800df5e) to the PR.

Reply:
you can use this one to get 3 Azs..2 Public Subnet...Privates and Intra...if you just need 3 Public or whatever you can use that  https://github.com/redhat-developer/mapt/blob/main/pkg/provider/aws/modules/network/standard/standard.go#L46 network request with your expected values...i.e only the PublicSubnet blocks...the create and that should do the trick

Reply:
Ah, yes, this works great 😄 Thank you! I got it working using that with ease. It should've caught my eye 😓